testing/test-utils/src/lib/utils/nx-plugin.ts

```diff
@@ -0,0 +1,57 @@
+import { CreateNodes, CreateNodesContext, CreateNodesResult } from '@nx/devkit';
+import { vol } from 'memfs';
+import { MEMFS_VOLUME } from '../constants';
+
+/**
+ * Unit Testing helper for the createNodes function of a Nx plugin.
+ * This function will create files over `memfs` from testCfg.matchingFilesData
+ * and invoke the createNodes function on each of the files provided including potential createNodeOptions.
+ * It will aggregate the results of each invocation and return the CreateNodesResult.
+ *
+ * @example
+ * ```ts
+ * const {projects = {}} = await createFilesAndInvokeCreateNodesOnThem(createNodes, context, undefined, { matchingFilesData});
+ *  // project should have one target created
+ *  const targets = projects[projectRoot]?.targets ?? {};
+ *  expect(Object.keys(targets)).toHaveLength(1);
+ *  // target should be the init target
+ *  expect(targets[`${CP_TARGET_NAME}--init`]).toBeDefined();
+ *  ```
+ *
+ * @param createNodes
+ * @param context
+ * @param createNodeOptions
+ * @param mockData
+ */
+export async function createVirtualFilesAndInvokeCreateNodesOnThem<
```

Comment:
Although I appreciate the descriptiveness, this function name is a real mouthful :sweat_smile: Since there's also an extensive description in JSDocs, I think it's fine to shorten the name a little, e.g.:

```suggestion
export async function invokeCreateNodesOnVirtualFiles<
```

Reply:
Happily change it back!